### PR TITLE
S122 lifecycle value ignore

### PR DESCRIPTION
**Pull request by jlorper**
- Lifecycle ignore for certain parameters
- Fail on error when building jar
- Update tag

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

File: infra/examples-dev/aws-msft-365/main.tf [hunk 1]
```diff
@@ -156,6 +162,12 @@ resource "aws_ssm_parameter" "refresh_endpoint" {
   type      = "String"
   overwrite = true
   value     = "https://login.microsoftonline.com/${var.msft_tenant_id}/oauth2/v2.0/token"
+
+  lifecycle {
+    ignore_changes = [
+      value
+    ]
+  }
```
Comment (eschultink) [inline infra/examples-dev/aws-msft-365/main.tf:170]: I don't think you want this here ...

File: infra/examples/aws-msft-365/main.tf [hunk 2]
```diff
@@ -156,13 +162,19 @@ resource "aws_ssm_parameter" "refresh_endpoint" {
   type      = "String"
   overwrite = true
   value     = "https://login.microsoftonline.com/${var.msft_tenant_id}/oauth2/v2.0/token"
+
+  lifecycle {
+    ignore_changes = [
+      value
+    ]
+  }
```
Comment (eschultink) [inline infra/examples/aws-msft-365/main.tf:170]: doesn't make sense here. this isn't something they change

**Review by aperez-worklytics — APPROVED** [review]

File: infra/modules/aws/main.tf [hunk 3]
```diff
@@ -120,6 +120,12 @@ resource "aws_ssm_parameter" "salt" {
   type        = "SecureString"
   description = "Salt used to build pseudonyms"
   value       = sensitive(random_password.random.result)
+
+  lifecycle {
+    ignore_changes = [
+      value
+    ]
+  }
```
Comment (eschultink) [inline infra/modules/aws/main.tf:128]: OK, although a bit weird.  should certainly comment this. I think it's entirely legitimate for us to pre-generate a value, and then they overwrite it if they don't trust ours - and if so, our terraform doesn't replace it - but still should comment that that's what's happening.

**Review by eschultink — APPROVED** [review]

**Comment by eschultink** [comment]
tech debt ticket to follow-up: https://app.asana.com/0/1202204100399673/1202256552566197

**Pull request merged**